Add initial release testing for outerloop and jitstress pipelines

trylek opened the pull request:
I have fixed the CoreCLR outerloop pipeline to include the release
branches. For jitstress and libraries_jitstress I have split the
schedules to run every other day for master vs. release as with
the growing number of Mono runs there's not much space for maneuver
w.r.t. allocating new lab HW.

Fixes: https://github.com/dotnet/runtime/issues/41841

Thanks

Tomas

P.S. There's not much I can do about validating that the updated
schedules actually work apart from just running the pipeline on the
PR to at least verify that the YAML is syntactically correct.

/cc: @dotnet/runtime-infrastructure

trylek committed:
Add initial release testing for outerloop and jitstress pipelines

I have fixed the CoreCLR outerloop pipeline to include the release
branches. For jitstress and libraries_jitstress I have split the
schedules to run every other day for master vs. release as with
the growing number of Mono runs there's not much space for maneuver
w.r.t. allocating new lab HW.

Fixes: https://github.com/dotnet/runtime/issues/41841

Thanks

Tomas

hoyosjs reviewed (approved):
Thanks Tomas :) I let the JIT team decide if they are ok with the alternating coverage. We would need to port this, right?

BruceForstall reviewed (approved):
Looks like you could add Sunday runs to the cron for master or 5.0; currently you're not doing any run on Sunday -- previously it was every day.

safern reviewed (approved):
LGTM, if we want this to work on release branches I believe we need to port it to them.

trylek commented:
@safern - I must admit I'm somewhat confused here. When I click "Edit" on the "runtime-coreclr jitstress" pipeline definition, for instance, it claims to be using a YAML script off the "master" branch. Are you sure we need to port the change or should we loop in more people to help us clarify this?

safern commented:
That is the "default" branch where a build would be queued and the yml that would be used. However, when you run a build from a PR branch or a CI branch, it will use the yml that is checked in on that branch.

trylek committed:
Add Sunday runs for [libraries-]jitstress per Bruce's PR feedback

I have somewhat arbitrarily added a Sunday run for jitstress on
release branches and for libraries-jitstress on master; I don't
think I have any compelling reason for not doing it the other way
round, except perhaps for a bit of symmetry in the YAML code.

Thanks

Tomas

trylek commented:
Thanks Santi for clarifying. I'll prepare a porting request after merging this in.

trylek commented:
/backport to release/5.0-rc2
